Makefile

```diff
@@ -410,60 +427,55 @@ tools/$(HTML_COMPRESSOR):
-	$(Q) cp -r html/wifi/*.png html_compressed/wifi;
-	$(Q) cp -r html/wifi/*.js html_compressed/wifi;
+$(BUILD_BASE)/espfs_img.o: $(shell find html) espfs/mkespfsimage/mkespfsimage
+	$(Q) rm -rf html_compressed; mkdir -p html_compressed; cp -r html/* html_compressed; 
 ifeq ("$(COMPRESS_W_HTMLCOMPRESSOR)","yes")
-	$(Q) echo "Compression assets with htmlcompressor. This may take a while..."
-		$(Q) java -jar tools/$(HTML_COMPRESSOR) \
-		-t html --remove-surrounding-spaces max --remove-quotes --remove-intertag-spaces \
-		-o $(abspath ./html_compressed)/ \
-		$(HTML_PATH)head- \
-		$(HTML_PATH)*.html
-	$(Q) java -jar tools/$(HTML_COMPRESSOR) \
-		-t html --remove-surrounding-spaces max --remove-quotes --remove-intertag-spaces \
-		-o $(abspath ./html_compressed)/wifi/ \
-		$(WIFI_PATH)*.html
-	$(Q) echo "Compression assets with yui-compressor. This may take a while..."
+	$(Q) echo "Compression assets with htmlcompressor. This is fast..."
+	$(Q) for file in `find html_compressed -type f -name "*.html"`; do \
+			java -jar yui/$(HTML_COMPRESSOR) \
+			-t html --remove-surrounding-spaces max --remove-quotes --remove-intertag-spaces \
+			-o $$file $$file; \
+		done
+	$(Q) echo "Compressing JS/CSS with yui-compressor. This may take a while..."
 	$(Q) for file in `find html_compressed -type f -name "*.js"`; do \
-			java -jar tools/$(YUI_COMPRESSOR) $$file --line-break 0 -o $$file; \
+			java -jar yui/$(YUI_COMPRESSOR) $$file --line-break 0 -o $$file; \
 		done
 	$(Q) for file in `find html_compressed -type f -name "*.css"`; do \
-			java -jar tools/$(YUI_COMPRESSOR) $$file -o $$file; \
-		done
-else
-	$(Q) cp -r html/head- html_compressed;
-	$(Q) cp -r html/*.html html_compressed;
-	$(Q) cp -r html/wifi/*.html html_compressed/wifi;	
+			java -jar yui/$(YUI_COMPRESSOR) $$file -o $$file; \
+		done 
 endif
 ifeq (,$(findstring mqtt,$(MODULES)))
 	$(Q) rm -rf html_compressed/mqtt.html
 	$(Q) rm -rf html_compressed/mqtt.js
 endif
-	$(Q) for file in `find html_compressed -type f -name "*.htm*"`; do \
-		cat html_compressed/head- $$file >$${file}-; \
-		mv $$file- $$file; \
-	done
-	$(Q) rm html_compressed/head-
+	$(Q) echo "Now building espFS ..."
 	$(Q) cd html_compressed; find . \! -name \*- | ../espfs/mkespfsimage/mkespfsimage > ../build/espfs.img; cd ..;
-	$(Q) ls -sl build/espfs.img
+	@echo "espFS image is $$(stat -c '%s' build/espfs.img) bytes =" $(shell expr $$(stat -c '%s' build/espfs.img) \* 100 / $$(du -sb html_compressed | { read first _ ; echo $$first; })) "% of uncompressed originals"
 	$(Q) cd build; $(OBJCP) -I binary -O elf32-xtensa-le -B xtensa --rename-section .data=.espfs \
-			espfs.img espfs_img.o; cd ..
+			espfs.img espfs_img.o; cd .. 
 
 # edit the loader script to add the espfs section to the end of irom with a 4 byte alignment.
 # we also adjust the sizes of the segments 'cause we need more irom0
+# in the end the only thing that matters wrt size is that the whole shebang fits into the
+# 236KB available (in a 512KB flash) 
```

Comment:
"ls -sl build/espfs.img" is of doubtful use.
Showing how many bytes take compressed assets in comparison with the uncompressed will be even more useful if you show this number as a percentage - it is much easier for a human to comprehend 2-digit percentage than 5 digits numbers in bytes.